### ISSUE TITLE
A/D navigator for parent relationships follows wrong direction

### ISSUE THREAD

**Issue by nonamethanks** [danbooru/danbooru#4771]
Consider https://danbooru.donmai.us/posts/4417794?q=parent%3A4417794

You'd expect D to go to the next post, but it's actually A.

**Comment by Thayol** [comment]
Take [post #4241063](https://danbooru.donmai.us/posts/4241063?q=parent%3A4241063) for example. 
Parents don't have a custom order, they follow ids. I would consider this as a feature request rather than a bug.

This behavior is working as you would expect from `show_seq`, the method that handles the search navbar.

We would have to extend `post_navbar_component.html.erb` with a completely new "Parent" navbar that is automatically added under every post that has a parent/child/sibling.

**Comment by nonamethanks** [comment]
Your example only works correctly because the child was uploaded before, as the parent is a revision. If it had been uploaded after, it'd have an inverted direction just like my example.

**Comment by Thayol** [comment]
Which is exactly why I would label this as a feature.

Relationships cannot go in the "wrong direction" because they don't have a direction. In the original example, the "next post" is chronologically speaking the next post, which doesn't exist in the context of the search `parent:4417794`.

Adding a "navbar for parent relationships" would also save a click when viewing the post without a query pointing to the parent.

**Comment by luni3359** [comment]
Maybe we could have a feature to override post order, that way we can also fix children order displayed at the top when somebody has uploaded pictures in the wrong order.

**Comment by Thayol** [comment]
@luni3359 Good idea. Something like "child priority" that could be used to order children would be good, but I don't know how wise it would be to add a completely new column to the database. Open a new issue for it if you have a good description of the feature or it's going to be forgotten.

**Comment by nonamethanks** [comment]
A rework of post relationships and pools is planned but it would be a massive amount of work, so it'll probably be a long time before it's done.

**Comment by luni3359** [comment]
If it's planned should I still make an issue about it?
